packages/launch/src/hooks/use-plans.tsx

```diff
@@ -21,4 +30,35 @@ export const usePlans = function usePlans() {
 	const planPrices = useSelect( ( select ) => select( PLANS_STORE ).getPrices( '' ) );
 
 	return { defaultPaidPlan, defaultFreePlan, planPrices };
-};
+}
+
+export function usePlanProductFromCart(): PlanProduct | undefined {
+	const { siteId } = React.useContext( LaunchContext );
+	const { getCart } = useDispatch( SITE_STORE );
+
+	const [ planProductFromCart, setPlanProductFromCart ] = React.useState< PlanProduct | undefined >(
+		undefined
+	);
+
+	React.useEffect( () => {
+		( async function () {
+			const cart = await getCart( siteId );
+			const planProduct = cart.products?.find( ( item: Product ) => isPlanProduct( item ) );
+			setPlanProductFromCart( planProduct );
+		} )();
+	}, [ siteId, getCart, setPlanProductFromCart ] );
+
+	return planProductFromCart;
+}
```

Comment:
Suggested a wp-data friendly approach in https://github.com/Automattic/wp-calypso/pull/47999/files#diff-e7e38dec68b1a4011df8cce1217ea7c0554561fca279540153d2216f961b97f5R46-R53

Current solution works fine and re-uses what we have so it's not a blocker for this PR.

Reply:
Okay gotcha!